src/main/webapp/resources/js/pages/user/components/UserDetailsPage.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from "react";
+import { useParams } from "react-router-dom";
+import { Button, Form, Input, notification, Select, Space, Switch, Typography } from 'antd';
+import moment from 'moment';
+
+import { useGetUserDetailsQuery, useEditUserDetailsMutation } from "../../../apis/users/users";
+
+/**
+ * React component to display the user details page.
+ * @returns {*}
+ * @constructor
+ */
+export default function UserDetailsPage() {
+  const { userId } = useParams();
+  const { data, isSuccess } = useGetUserDetailsQuery(userId);
+  const [editUser] = useEditUserDetailsMutation();
+
+  const [formErrors, setFormErrors] = useState();
+
+  const onFormFinish = (values) => {
+    editUser({'userId': userId, ...values})
+      .unwrap()
+      .then((payload) => {
+        notification.success({ message: i18n("UserDetailsPage.notification.success") });
+      })
+      .catch((error) => {
+        notification.error({ message: i18n("UserDetailsPage.notification.error") });
+        setFormErrors(error.data);
+      });
+  }
+
+  if (isSuccess) {
+    return (
+      <Space direction="vertical">
+        <Typography.Title level={4}>{data.user.username}</Typography.Title>
+        <Form
+          layout="vertical"
+          initialValues={data.user}
+          onFinish={onFormFinish}
+        >
+          <Form.Item label={i18n("UserDetailsPage.form.label.firstName")} name="firstName" help={formErrors?.firstName} validateStatus={ formErrors?.firstName ? "error" : undefined }>
```

Comment:
The formatting needs to be fixed on these form items. Do you have prettier installed?

Reply:
Please see 8dc9fb77580f3757ecd93002e0f5f56cf610397b.